### PR TITLE
Introduce an arguments descriptor that says keyword arguments are present

### PR DESCRIPTION
**Pull request by chrisseaton**
More minimal version of #2597.

### CODE REVIEW

**Comment by eregon** [comment]
Self-note: the main way to review correctness is to audit usages of `EmptyArgumentsDescriptor.INSTANCE`

File: src/main/java/org/truffleruby/parser/MethodTranslator.java [hunk 1]
```diff
@@ -490,7 +490,7 @@ public RubyNode visitZSuperNode(ZSuperParseNode node) {
                 reloadSequence.getSequence());
         final RubyNode block = executeOrInheritBlock(argumentsAndBlock.getBlock(), node);
 
-        RubyNode callNode = new SuperCallNode(arguments, block);
+        RubyNode callNode = new SuperCallNode(arguments, block, null);
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/MethodTranslator.java:493]: Why not `argumentsAndBlock.getArgumentsDescriptor()`? Whether `super` passes kwargs depends on whether the method calling super declares kwargs, right?

Comment (chrisseaton) [reply src/main/java/org/truffleruby/parser/MethodTranslator.java:493]: `argumentsAndBlock.getArgumentsDescriptor()` is the static information about the `super` call site. We don't have static information about the call site of the caller of the method containing the super. Remember descriptors are call-site information. So this `null` prompts it to do it dynamically.

Comment (eregon) [reply src/main/java/org/truffleruby/parser/MethodTranslator.java:493]: I found a way, we can decide like this:
```java
        final ArgumentsDescriptor descriptor = argsNode.hasKwargs()
                ? KeywordArgumentsDescriptor.INSTANCE
                : EmptyArgumentsDescriptor.INSTANCE;
```
(and argsNode is the surrounding method's parameters)

**Pull request merged**